src/api/java/com/minecolonies/api/util/InventoryUtils.java

```diff
@@ -874,14 +874,27 @@ public static boolean addItemStackToItemHandler(@NotNull final IItemHandler item
                 slot = itemHandler.getSlots() == 0 ? -1 : 0;
                 while (!ItemStackUtils.isEmpty(resultStack) && slot != -1 && slot != itemHandler.getSlots())
                 {
-                    resultStack = itemHandler.insertItem(slot, resultStack, false);
+                    resultStack = itemHandler.insertItem(slot, resultStack, true);
                     if (!ItemStackUtils.isEmpty(resultStack))
                     {
                         slot++;
                     }
+                    // result empty, we can insert the whole stack
```

Comment:
and if we were able to insert a partial stack first and then the rest?
This should have worked fine up to now.

Reply:
resultStack is the remainder already for the partial insert, which gets re-used in the loop till its empty or the loop failed. So when its empty we can fit the whole. But can probably simplify this a lot later.

The issue with this was that the return value lost information,  doing a partial insert combined with true/false return either leads to some information loss, or to having to modify the stack parameters size and hoping there is followup logic which is really hacky. So this is now all or nothing

Reply:
This is flawed logic. There are inventories from which you can't extract and not insert.

Reply:
you're saying there are inventories which take a not-empty stack into insert and then return an empty stack without beeing able to insert it?

Reply:
Yes

Reply:
hm but I think if insert with simulate works then insert without simulate should work aswell?